File: ws/Vaca/u2/WS23UnitTest/ComplexOperations/ComplexOperations/src/test/java/ec/edu/espe/complexoperations/controller/PersonControllerTest.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/UnitTests/JUnit5TestClass.java to edit this template
 */
package ec.edu.espe.complexoperations.controller;

import java.time.LocalDate;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
 *
 * @author Luis Vaca, TheJavaBandits, DCCO-ESPE
 */
public class PersonControllerTest {
    
    public PersonControllerTest() {
    }

    @Test
    public void testComputeAgeInDays() {
        System.out.println("computeAgeInDays");
        LocalDate birthDate = LocalDate.parse("2005-02-26");
        int expResult = 7072;
        int result = PersonController.computeAgeInDays(birthDate);
        assertEquals(expResult, result);
    }

    @Test
    public void testComputeAgeInYears() {
        System.out.println("computeAgeInYears");
        LocalDate birthDate = LocalDate.parse("2005-02-26");
        int expResult = 19;
        int result = PersonController.computeAgeInYears(birthDate);
        assertEquals(expResult, result);
    }
    
}

File: hw/Pérez/U2/HW17TestCaseSolution/Test/PersonController.test.js
import PersonController from '../ec.edu.espe.complexoperations.controller/PersonController.js';

test('computeAgeInDays returns correct age in days', () => {
    const birthDate = '2003-02-22';
    const result = PersonController.computeAgeInDays(birthDate);
    expect(result).toBe(7807); 
});

test('computeAgeInYears returns correct age in years', () => {
    const birthDate = '2003-02-22';
    const result = PersonController.computeAgeInYears(birthDate);
    expect(result).toBe(21); 
});
